### PR TITLE
fix: adding or modifying a nested property requires addField permissions

### PR DESCRIPTION
**Pull request by bdevore17**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
This PR is the rebased version of #7372 

Related issue: #7372 and #7371 

### Approach
See #7371 
Also added a corresponding regression test

### TODOs before merging

- [x] Add tests
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: spec/schemas.spec.js [hunk 1]
```diff
@@ -1779,6 +1779,34 @@ describe('schemas', () => {
...
+      testSchema.setCLP({
+        create: { ['*']: true },
+        update: { ['*']: true },
+        addField: {},
+      });
+      testSchema.addObject('a');
+      await testSchema.save();
+    });
+
+    it('addField not required for adding a nested field (#7371)', async () => {
```
Comment (mtrezza) [inline spec/schemas.spec.js]: could you please rename to:

> it('addField permission not required for adding a nested property', async () => {

File: spec/schemas.spec.js [hunk 2]
```diff
@@ -1779,6 +1779,34 @@ describe('schemas', () => {
...
+      const obj = new Parse.Object('test_7371');
+      obj.set('a', {});
+      await obj.save();
+      obj.set('a.b', 2);
+      await obj.save();
+    });
+    it('addField not required for modifying a nested field (#7371)', async () => {
```
Comment (mtrezza) [inline spec/schemas.spec.js]: could you please rename to:

> it('addField permission not required for modifying a nested property', async () => {

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by bdevore17** [comment]
@mtrezza done! Let me know if you need anything else here.

**Comment by bdevore17** [comment]
brought this up to date with alpha^

**Review by mtrezza — APPROVED** [review]
Looks good! Thanks for the fix. Let's just wait for the CI to pass.

**Comment by bdevore17** [comment]
@mtrezza What do I need to do here? It doesn't seem like any of these failures relate to this PR.

**Comment by mtrezza** [comment]
@bdevore17 could you just bring this branch up-to-date to see if the tests pass? We have merged some fixes on the alpha branch.

**Comment by bdevore17** [comment]
@mtrezza done!

**Comment by mtrezza** [comment]
Restarted the CI, let's see whether it passes.

**Comment by bdevore17** [comment]
@mtrezza looks like it passed. Can this now be merged?

**Comment by mtrezza** [comment]
Could you update the branch please?

**Comment by bdevore17** [comment]
@mtrezza done!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-alpha.12](https://github.com/parse-community/parse-server/releases/tag/5.0.0-alpha.12)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.10](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.10)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0](https://github.com/parse-community/parse-server/releases/tag/5.1.0)

**Pull request merged**